### PR TITLE
Move transaction futures into their own modules

### PR DESCRIPTION
**Pull request by nlordell**
The transaction module was over 1000 lines and was getting hard to navigate.

### CODE REVIEW

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FxT5LMPcAaKKiOzwGsw%2Fgiphy.gif&token=move" data-gitmeme-token="move"></a>

File: src/transaction.rs [hunk 1]
```diff
@@ -70,6 +63,37 @@ impl Default for ResolveCondition {
...
+pub enum Transaction {
+    /// A structured transaction request to be signed locally by the node.
+    Request(TransactionRequest),
+    /// A signed raw transaction request.
+    Raw(Bytes),
+}
+
+impl Transaction {
+    /// Unwraps the transaction into a transaction request, returning None if the
+    /// transaction is a raw transaction.
```
Comment (bh2smith) [inline src/transaction.rs:79]: Don't all successful transactions return a receipt of some type (regardless of whether we were expecting a return value).

**Review by bh2smith — APPROVED** [review]
Once again, this all looks really great. I am impressed by all the dedication to doc-strings, but I am unsure if I can provide a proper review here. There is so much going on.

File: src/transaction.rs (hunk 1 above)
Comment (nlordell) [reply src/transaction.rs:79]: This is for sending transactions, they can either be a transaction object (which is unsigned and requires the node to sign transactions locally) or a raw transactions (a bunch of bytes, that are signed). This utility method just returns `Some(TransactionRequest)` if it is the former, and `None` if it is the latter.

**Pull request merged**